src/platform/Darwin/BleConnectionDelegateImpl.mm

```diff
@@ -303,9 +331,22 @@ - (void)dispatchConnectionError:(CHIP_ERROR)error
 - (void)dispatchConnectionComplete:(CBPeripheral *)peripheral
 {
     auto * onConnectionComplete = _onConnectionComplete;
+    auto * onConnectionCompleteWithDiscriminator = _onConnectionCompleteWithDiscriminator;
+    auto * matchedLongDiscriminator = _matchedLongDiscriminator;
+    auto * onConnectionError = _onConnectionError;
     auto * appState = _appState;
     [self clearConnectionCallbacks];
-    if (onConnectionComplete != nullptr) {
+    if (onConnectionCompleteWithDiscriminator != nullptr) {
+        if (matchedLongDiscriminator == nil) {
+            // Very unexpected!  We should only have a peripheral without a
```

Comment:
Given we enforce this invariant internally in this class, I think this can just be a VerifyOrDie. Calling onConnectionError (without a null check too) doesn't seem like the right thing to do

Reply:
Done, with VerifyOrDie.